### PR TITLE
fix(security): the dependency scan stops at the repository's own manifests

### PR DESCRIPTION
**Pull request by piwi3910**
## What

Scope dependency scanning and npm/Python gap checks to tracked and untracked-but-not-ignored files, excluding deleted paths and vendored dependencies.

## Why

Ignored agent installs such as .kilocode/package-lock.json must not block commits for dependencies the repository did not choose. Empty Git inventories, deleted tracked paths, and filesystem-only gap checks left holes in that boundary.

## How

Shared dependencyFiles discovery preserves Git success separately from count, uses NUL-delimited paths, excludes missing files and directory replacements, and retains the non-Git filesystem fallback. Gap checks and sibling/workspace lockfile coverage use the same ownership rules.

## Testing

procoder test passed 57 Go packages; standalone JavaScript NOT run because no test script exists. procoder check: 4 clean files and zero blocking findings. Added regressions for empty inventories, deleted files, quoted filenames, tracked/untracked/vendor scope, ignored gap manifests, and ignored coverage lockfiles. Existing non-Git monorepo coverage passes. Adversarial and edge-case review completed. CI run 35271369954 passed gate and all three OS test jobs; CodeQL checks also passed.

## Notes for the reviewer

Review fixes are in 9cdaf177c6979fca5f9727d89c575aa0f85d1179. Both review threads have evidence replies and are resolved; the two additional review observations are addressed too. #289 is stacked on this branch and must be retargeted to main after this dependency merges.

### CODE REVIEW

File: internal/security/security.go [hunk 1]
```diff
@@ -981,6 +996,27 @@ func manifestsIn(root string) []string {
...
 	}
+	// Gitignored manifests are not this repository's dependencies. Agent
+	// tooling installs into the working tree (.kilocode/, .claude/ and
+	// friends each ship a package-lock.json), and scanning those reports
+	// vulnerabilities in packages nobody here chose, pinned, or can
+	// upgrade — while blocking every commit until they do. The gate's own
+	// file set is the right scope, and it is the scope `procoder audit`
+	// already claims out loud. Falls through to the walk when git cannot
+	// answer, so a non-repo directory still gets scanned.
+	if tracked := gitx.FilesUnder(root, "."); len(tracked) > 0 {
```
[the remark this answers is not in the document]
Comment (piwi3910) [reply internal/security/security.go]: Fixed in 9cdaf17. dependencyFiles now branches on the Git command error, not the inventory length, and NUL-delimits paths. TestEmptyGitInventoryDoesNotFallBackToIgnoredManifests creates a repository whose ignore file and manifests are all ignored and verifies scan/npm/Python inventories remain empty. procoder test passed all 57 Go packages; procoder check passed with zero blocking findings.

File: internal/security/security.go [hunk 2]
```diff
@@ -981,6 +996,27 @@ func manifestsIn(root string) []string {
...
+		var out []string
+		for _, p := range tracked {
+			if !names[filepath.Base(p)] || inManifestDir(root, p) {
+				continue
+			}
```
[the remark this answers is not in the document]
Comment (piwi3910) [reply internal/security/security.go:988]: Fixed in 9cdaf17. Discovery stats each candidate and excludes missing paths and directory replacements while preserving other errors for scanner reporting. TestDeletedTrackedManifestsAreNotScanned stages a lockfile, removes it, then replaces it with a directory; neither state enters the scanner inventory. procoder test passed all 57 Go packages; procoder check passed with zero blocking findings.

**Comment by piwi3910** [comment]
Addressed the two additional review observations in 9cdaf17: the original fixture now stages go.mod and a vendored manifest while keeping a real nested lockfile untracked; npmGaps and pythonGaps share dependencyFiles with scanner discovery. Added tracked/untracked/ignored/vendor gap coverage and fixed ignored sibling/workspace lockfiles suppressing real gaps. Git-quoted paths now use NUL separation with a regression. Non-Git fallback remains covered. Full procoder test: 57 Go packages passed; standalone JS NOT run (no test script). procoder check: 4 clean files, zero blocking findings. Adversarial and edge-case review passes completed; CI is running on the new head.
